polyhost/device/poly_kybd_mock.py

```
@@ -154,39 +172,86 @@ def send_overlays(self, filenames):
         ov = mapping[keycode]
+        pool_slot = display_flat_idx(keycode, modifier)
+        self._sim.store_image(pool_slot, ov.all_bytes)
         return min(ov.all_msgs, ov.compressed_msgs, ov.roi_msgs, ov.compressed_roi_msgs)
 
+    # ── inspection helpers ──────────────────────────────────────────────────
+
+    def get_display_bitmap(self, keycode: int, modifier: Modifier) -> bytes | None:
+        """Return the 360-byte bitmap shown at (keycode, modifier), or None if blank."""
+        return self._sim.get_display_bitmap(keycode, modifier)
+
+    def get_display_image(self, keycode: int, modifier: Modifier) -> "np.ndarray | None":
```

Ruff (0.15.12)

[error] Undefined name `np`

(F821)